storage/src/main/java/zipkin2/storage/kafka/KafkaSpanStore.java

```diff
@@ -204,6 +216,29 @@
     }
   }
 
+  static final class GetTraceManyCall extends KafkaStoreScatterGatherListCall<List<Span>> {
+    final KafkaStreams traceStoreStream;
+    final BiFunction<String, Integer, String> httpBaseUrl;
+    final String traceIds;
+
+    GetTraceManyCall(KafkaStreams traceStoreStream,
+      BiFunction<String, Integer, String> httpBaseUrl,
+      String traceIds) {
+      super(traceStoreStream, TRACES_STORE_NAME, httpBaseUrl, "/traceMany?traceIds=" + traceIds);
+      this.traceStoreStream = traceStoreStream;
+      this.httpBaseUrl = httpBaseUrl;
+      this.traceIds = traceIds;
+    }
+
+    @Override protected List<Span> parse(JsonNode node) throws JsonProcessingException {
```

Comment:
is this right? trace many should each return a list of traces..

Reply:
`parse` is used for individual elements, in this case List<Span>, that will be listed later as a List<List<Span>>, `parseItem` + comment should be better; but actually this make me realize we can optimize this call:

Currently if we have `n` traceIds, and `m` instances, we are doing `m` calls, which is not efficient when `m`>`n`.

But with with RPC metadata we can call `n` times, as we know where each trace is, reducing from `m` to `n`, when `n`<`m`

And combining both approaches, i.e. get host by traceId first instead of on the http service and then call each host, we can reduce calls to eventually one when traceIds are on the same host.

Will work on the refactoring

Reply:
great explanation. fire when ready